### PR TITLE
Code and comment fixes for stream

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
Jenkins, test this please.

File: cupy/cuda/stream.pyx [hunk 1]
```diff
@@ -411,14 +416,13 @@ class ExternalStream(BaseStream):
...
-        self.ptr = ptr
         # It is in theory unsafe to just call runtime.getDevice() here, as the
         # stream pointer could come from a different device (although
         # unlikely). While we could use driver API combos cuStreamGetCtx ->
         # cuCtxSetCurrent -> cuCtxGetDevice -> ... to retrieve the device ID
-        # associated with the stream, it is way too complicated. Let us keep
-        # this as thin as possible.
-        self.device_id = device_id
+        # associated with the stream, it is way too complicated and does not
+        # work with HIP. Let us keep this as thin as possible.
```
Comment (leofang) [inline cupy/cuda/stream.pyx:424]: This is referring to https://github.com/cupy/cupy/pull/5172#discussion_r625841360. Thanks for adding it!

**Review by leofang — APPROVED** [review]
Thanks, LGTM. (Looks like the fix scope is reduced? IIRC the version I read this morning included other nontrivial fixes.)

**Comment by kmaehashi** [comment]
Yes, the original motivation of this refactoring was to improve weakref handling, but as weakref will be removed in #5251 this PR now became just a code cleanup 😅

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**